database/sqlite3-driver/lite3_database.ml

```diff
@@ -94,6 +94,13 @@ class lite3_database file = object(self)
   method exec query : Value.dbvalue =
     Debug.print query;
     let stmt = prepare connection query in
+    let rec last_res r =
```

Comment:
Yes this can be considered Sqlite3 boilerplate. The manual essentially states that calling `prepare` may not execute all of the command, and to determine if this is the case `prepare_tail` can be called on the resulting statement. It returns `None` if the computation is finished, otherwise `Some x`, where `x` is a new statement. This is used to ensure that multiple statements are executed.

Reply:
Yes. this is discussed further in #894.  I'll add a comment documenting this.  It would probably be good practice to qualify the Sqlite3 library imports too.